src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
+
+export const APIResponse = {
+  create: (
+    isSuccess: number,
+    statusCode: number,
+    message?: string,
+    data?: any,
+    error?: any,
+  ): ControllerResponse => {
+    const response: ControllerResponse = { isSuccess, statusCode, message };
+    if (error) {
+      response.error = error;
+    } else if (!error && data) {
+      response.data = data;
+    } else {
+      response.data = {};
+    }
```

Comment:
PullProofAI

The `APIResponse` object in the `apiResponse.utils.ts` file seems well-structured and handles creating controller responses effectively. However, there are a few improvements that can be suggested:

1. It's good practice to avoid using `any` type whenever possible to maintain type safety.
2. Consider using interfaces instead of types for better readability and maintainability.
3. The logic for setting `data` and `error` fields can be simplified.

Here are the recommended changes:

```diff
/* eslint-disable @typescript-eslint/no-explicit-any */
export interface ControllerResponse {
  isSuccess: number;
  statusCode: number;
  message: string;
  data?: any;
  error?: any;
}

export const APIResponse = {
  create: (
    isSuccess: number,
    statusCode: number,
    message?: string,
    data?: any,
    error?: any,
  ): ControllerResponse => {
    const response: ControllerResponse = { isSuccess, statusCode, message };
    
    if (error) {
      response.error = error;
    } else {
      response.data = data || {};
    }

    return response;
  },
};
```

By making these changes, the code will be more robust, easier to understand, and maintain in the long run.


<!-- This is an auto-generated comment by PullProofAI -->